zerver/lib/events.py

```diff
@@ -960,6 +960,8 @@ def _draft_update_action(i: int) -> None:
                 # Then there are some extra fields that also need to be set.
                 state["zulip_plan_is_not_limited"] = event["value"] != Realm.PLAN_TYPE_LIMITED
                 state["realm_upload_quota_mib"] = event["extra_data"]["upload_quota"]
+                if event["value"] == Realm.PLAN_TYPE_LIMITED:
+                    state["realm_enable_spectator_access"] = False
```

Comment:
I think we'd get to cut this line if we do the `do_set_realm_property` approach noted above, which is one reason I kinda want to do that.